src/main/ThemeSwitcher.tsx

```diff
@@ -14,14 +14,15 @@ const ThemeSwitcher: React.FC<{}> = () => {
 
   React.useEffect(() => {
     localStorage.setItem('theme', themeState)
-    dispatch(setTheme(theme))
-    // Listen to system preference changes
-    const isDarkPrefered = window.matchMedia('(prefers-color-scheme: dark)');
-    isDarkPrefered.addEventListener('change', () => {
-      dispatch(setTheme(theme))
-    })
+    dispatch(toggleTheme(theme))
   }, [ themeState, theme, dispatch ] )
 
+  // Listen to system preference changes
+  const isDarkPrefered = window.matchMedia('(prefers-color-scheme: dark)');
+  isDarkPrefered.addEventListener('change', () => {
+    dispatch(toggleTheme(theme))
+  })
+
```

Comment:
Now the code is run on every rerender. Does this need to be run more than once on start-up?